# EX3: Some Statistics with Baby Names

### Introduction:

We are going to use a subset of [US Baby Names](https://www.kaggle.com/kaggle/us-baby-names) from Kaggle.  
In the file it will be names from 2004 until 2014


### Step 1. Import the necessary libraries

In [1]:
#import pandas as pd

import numpy as np
import pandas as pd

### Step 2. 
User ***Jupyter Lab*** to view ``../../DATA/US_Baby_Names_right.csv'``.

### Step 3. Assign it to a variable called baby_names.

In [88]:
baby_names=pd.read_csv('../../DATA/US_Baby_Names_right.csv')

### Step 4. See the first 10 entries

In [89]:
baby_names.head(10) #Erste 10 Einträge

Unnamed: 0     Id      Name  Year Gender State  Count
0       11349  11350      Emma  2004      F    AK     62
1       11350  11351   Madison  2004      F    AK     48
2       11351  11352    Hannah  2004      F    AK     46
3       11352  11353     Grace  2004      F    AK     44
4       11353  11354     Emily  2004      F    AK     41
5       11354  11355   Abigail  2004      F    AK     37
6       11355  11356    Olivia  2004      F    AK     33
7       11356  11357  Isabella  2004      F    AK     30
8       11357  11358    Alyssa  2004      F    AK     29
9       11358  11359    Sophia  2004      F    AK     28

### Step 5. Delete the column 'Unnamed: 0' and 'Id'

In [91]:
baby_names_adjusted = baby_names.drop(columns=['Unnamed: 0', 'Id']) #Frage: Warum wird die Methode drop nicht auf den Datensatz angewendet?

#del baby_names['Unnamed: 0']
#del baby_names['Id']

In [92]:
baby_names_adjusted

Name  Year Gender State  Count
0           Emma  2004      F    AK     62
1        Madison  2004      F    AK     48
2         Hannah  2004      F    AK     46
3          Grace  2004      F    AK     44
4          Emily  2004      F    AK     41
...          ...   ...    ...   ...    ...
1016390     Seth  2014      M    WY      5
1016391  Spencer  2014      M    WY      5
1016392     Tyce  2014      M    WY      5
1016393   Victor  2014      M    WY      5
1016394   Waylon  2014      M    WY      5

[1016395 rows x 5 columns]

### Step 6. Are there more male or female names in the dataset?

In [84]:
#Anzahl Rows in denen Gender = F
#len(baby_names_adjusted[baby_names_adjusted['Gender'] == 'F'])

#Anzahl Rows in denen Gender = M
#len(baby_names_adjusted[baby_names_adjusted['Gender'] == 'M'])

In [93]:
baby_names['Gender'].value_counts()

F    558846
M    457549
Name: Gender, dtype: int64

In [94]:
percent = baby_names['Gender'].value_counts(normalize=True)*100
print(percent[0], '% Anteil Frauennamen')
print(percent[1], '% Anteil Männernamen')

54.98315123549408 % Anteil Frauennamen
45.01684876450592 % Anteil Männernamen


### Step 7. Group the dataset by name and assign to names

In [95]:
#group by name 
names = baby_names_adjusted.groupby(baby_names_adjusted['Name'])
names.count() #number of group entries by columns

Year  Gender  State  Count
Name                               
Aaban       2       2      2      2
Aadan       4       4      4      4
Aadarsh     1       1      1      1
Aaden     196     196    196    196
Aadhav      1       1      1      1
...       ...     ...    ...    ...
Zyra        7       7      7      7
Zyrah       2       2      2      2
Zyren       1       1      1      1
Zyria      10      10     10     10
Zyriah      9       9      9      9

[17632 rows x 4 columns]

### Step 8. How many different names exist in the dataset?

In [96]:
names.ngroups

17632

Antwort: Im Datensatz existieren 17632 Namen

### Step 9. What is the name with most occurrences?

In [97]:
names['Count'].sum().sort_values()[-1:]

Name
Jacob    242874
Name: Count, dtype: int64

In [118]:
names.get_group('Jacob')

Name  Year Gender State  Count
2407     Jacob  2004      M    AK     54
2611     Jacob  2005      M    AK     58
2827     Jacob  2006      M    AK     55
3048     Jacob  2007      M    AK     49
3271     Jacob  2008      M    AK     60
...        ...   ...    ...   ...    ...
1015550  Jacob  2010      M    WY     21
1015708  Jacob  2011      M    WY     30
1015879  Jacob  2012      M    WY     28
1016056  Jacob  2013      M    WY     20
1016262  Jacob  2014      M    WY     13

[568 rows x 5 columns]

In [100]:
names['Count'].sum().max()

242874

In [101]:
names['Count'].agg([np.sum]).max()

sum    242874
dtype: int64

In [81]:
#sort by column and retun top n
#def topN(baby_names_adjusted, n=1, column='Count'):
#    return baby_names_adjusted.sort_values(by=column)[-n:]
#
#baby_names_adjusted.groupby(baby_names_adjusted['Name']).apply(topN)

### Step 10. How many different names have the least occurrences?

In [102]:
names['Count'].sum().sort_values()

Name
Destenie         5
Janisha          5
Lizvet           5
Arsalan          5
Janira           5
             ...  
Isabella    204798
Ethan       209277
Michael     214405
Emma        214852
Jacob       242874
Name: Count, Length: 17632, dtype: int64

In [105]:
names['Count'].sum()[names['Count'].sum() == names['Count'].sum().min()]

Name
Aadarsh     5
Aadin       5
Aaima       5
Aalaya      5
Aaminah     5
           ..
Zyien       5
Zyire       5
Zykeriah    5
Zykierra    5
Zymari      5
Name: Count, Length: 2578, dtype: int64

In [104]:
len(names['Count'].sum()[names['Count'].sum() == names['Count'].sum().min()])

2578

Antwort: Insgesamt 2578 Namen kommen am geringsten vor.

### Step 11. What is the median name occurrence?

In [107]:
names['Count'].sum().median()

49.0

### Step 12. What is the standard deviation of names?

In [108]:
names['Count'].agg([np.std])

std
Name              
Aaban     0.000000
Aadan     0.957427
Aadarsh        NaN
Aaden    21.154974
Aadhav         NaN
...            ...
Zyra      1.154701
Zyrah     0.707107
Zyren          NaN
Zyria     0.737865
Zyriah    1.666667

[17632 rows x 1 columns]

### Step 13. Get a summary with the mean, min, max, std and quartiles.

In [78]:
#Statistik für alle Namen
names['Count'].agg([np.min,np.max,np.mean,np.sum,np.std])

amin  amax       mean   sum        std
Name                                           
Aaban       6     6   6.000000    12   0.000000
Aadan       5     7   5.750000    23   0.957427
Aadarsh     5     5   5.000000     5        NaN
Aaden       5   158  17.479592  3426  21.154974
Aadhav      6     6   6.000000     6        NaN
...       ...   ...        ...   ...        ...
Zyra        5     8   6.000000    42   1.154701
Zyrah       5     6   5.500000    11   0.707107
Zyren       6     6   6.000000     6        NaN
Zyria       5     7   5.900000    59   0.737865
Zyriah      5    10   6.444444    58   1.666667

[17632 rows x 5 columns]

In [110]:
#Angaben mit quartiles
print(names['Count'].describe())

         count       mean        std  min   25%   50%    75%    max
Name                                                               
Aaban      2.0   6.000000   0.000000  6.0  6.00   6.0   6.00    6.0
Aadan      4.0   5.750000   0.957427  5.0  5.00   5.5   6.25    7.0
Aadarsh    1.0   5.000000        NaN  5.0  5.00   5.0   5.00    5.0
Aaden    196.0  17.479592  21.154974  5.0  6.00  10.0  20.00  158.0
Aadhav     1.0   6.000000        NaN  6.0  6.00   6.0   6.00    6.0
...        ...        ...        ...  ...   ...   ...    ...    ...
Zyra       7.0   6.000000   1.154701  5.0  5.00   6.0   6.50    8.0
Zyrah      2.0   5.500000   0.707107  5.0  5.25   5.5   5.75    6.0
Zyren      1.0   6.000000        NaN  6.0  6.00   6.0   6.00    6.0
Zyria     10.0   5.900000   0.737865  5.0  5.25   6.0   6.00    7.0
Zyriah     9.0   6.444444   1.666667  5.0  5.00   6.0   7.00   10.0

[17632 rows x 8 columns]
